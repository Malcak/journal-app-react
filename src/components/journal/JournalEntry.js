import React from 'react';

export const JournalEntry = () => {
  return (
    <div className="journal__entry pointer">
      <div
        className="journal__entry-picture"
        style={{
          backgroundSize: 'cover',
          backgroundImage:
            'url(https://miro.medium.com/max/1200/1*mk1-6aYaf_Bes1E3Imhc0A.jpeg)',
        }}
      ></div>
      <div className="journal__entry-body">
        <p className="journal__entry-title">Un nuevo d√≠a</p>
        <p className="journal__entry-content">
          Lorem ipsum dolor sit amet, consectetur adipisicing elit.
        </p>
      </div>
      <div className="journal__entry-date-box">
        <span>Monday</span>
        <h4>28</h4>
      </div>
    </div>
  );
};
