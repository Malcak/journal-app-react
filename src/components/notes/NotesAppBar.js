import React from 'react';

export const NotesAppBar = () => {
  return (
    <div className="notes__appbar">
      <span>28 de Agosto 2020</span>
      <div>
        <button className="btn">picture</button>
        <button className="btn">Save</button>
      </div>
    </div>
  );
};
