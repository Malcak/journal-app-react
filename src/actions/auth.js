import Swal from 'sweetalert2';
import { firebase, googleAuthProvider } from '../firebase/firebaseConfig';
import { types } from '../types/types';
import { notesLogoutClean } from './notes';

export const startLoginEmailPassword = (email, password) => {
  return (dispatch) => {
    dispatch({ type: types.uiStartLoading });
    return firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(({ user }) => {
        dispatch(login(user.uid, user.displayName));
        dispatch({ type: types.uiFinishLoading });
      })
      .catch((e) => {
        dispatch({ type: types.uiFinishLoading });
        Swal.fire('Error', e.message, 'error');
      });
  };
};

export const startRegisterWithEmailPasswordName = (email, password, name) => {
  return (dispatch) => {
    dispatch({ type: types.uiStartLoading });
    firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then(async ({ user }) => {
        await user.updateProfile({ displayName: name });
        dispatch(login(user.uid, user.displayName));
        dispatch({ type: types.uiFinishLoading });
      })
      .catch((e) => {
        dispatch({ type: types.uiFinishLoading });
        Swal.fire('Error', e.message, 'error');
      });
  };
};

export const startGoogleLogin = () => {
  return (dispatch) => {
    firebase
      .auth()
      .signInWithPopup(googleAuthProvider)
      .then(({ user }) => {
        dispatch(login(user.uid, user.displayName));
      })
      .catch((e) => {
        Swal.fire('Error', e.message, 'error');
      });
  };
};

export const login = (uid, displayName) => ({
  type: types.login,
  payload: {
    uid,
    displayName,
  },
});

export const startLogout = () => {
  return async (dispatch) => {
    await firebase.auth().signOut();
    dispatch(logout());
    dispatch(notesLogoutClean());
  };
};

export const logout = () => ({
  type: types.logout,
});
