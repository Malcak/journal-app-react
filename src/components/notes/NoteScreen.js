import React, { useEffect, useRef } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useForm } from '../../hooks/useForm';
import { NotesAppBar } from './NotesAppBar';
import { activeNote, startDeleting } from '../../actions/notes';

export const NoteScreen = () => {
  const dispatch = useDispatch();
  const { active: note } = useSelector((state) => state.notes);
  const [formValues, handleInputChange, reset] = useForm(note);
  const { title, body } = formValues;

  const activeId = useRef(note.id);

  useEffect(() => {
    if (note.id !== activeId.current) {
      reset(note);
      activeId.current = note.id;
    }
  }, [note, reset]);

  useEffect(() => {
    dispatch(activeNote(formValues.id, { ...formValues }));
  }, [formValues, dispatch]);

  const handleDelete = () => {
    dispatch(startDeleting(note.id));
  };

  return (
    <div className="notes__main-content">
      <NotesAppBar />
      <div className="notes__content">
        <input
          type="text"
          name="title"
          placeholder="Some awesome title!"
          className="notes__title-input"
          value={title}
          onChange={handleInputChange}
          autoComplete="off"
        />
        <textarea
          name="body"
          placeholder="What happened today..."
          className="notes__textarea"
          value={body}
          onChange={handleInputChange}
        ></textarea>
        {note.imgUrl && (
          <div className="notes__image">
            <img src={note.imgUrl} alt="noteImage" />
          </div>
        )}
      </div>
      <button className="btn btn-danger" onClick={handleDelete}>
        Remove
      </button>
    </div>
  );
};
