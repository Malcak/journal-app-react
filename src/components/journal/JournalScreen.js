import React from 'react';
import { NoteScreen } from '../notes/NoteScreen';
import { NothingSelected } from './NothingSelected';
import { Siderbar } from './Siderbar';

export const JournalScreen = () => {
  return (
    <div className="journal__main-content">
      <Siderbar />
      <main>
        {/* <NothingSelected /> */}
        <NoteScreen />
      </main>
    </div>
  );
};
