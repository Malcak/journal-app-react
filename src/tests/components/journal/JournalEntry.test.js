import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { JournalEntry } from '../../../components/journal/JournalEntry';
import { activeNote } from '../../../actions/notes';
import notes from '../../fixtures/notes';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const defaultState = {};

let store = mockStore(defaultState);
store.dispatch = jest.fn();

const wrapper = mount(
  <Provider store={store}>
    <MemoryRouter>
      <JournalEntry {...notes[0]} />
    </MemoryRouter>
  </Provider>
);

describe('test on <JournalEntry /> component', () => {
  test('should display correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should activate the note', () => {
    wrapper.find('.journal__entry').prop('onClick')();
    expect(store.dispatch).toHaveBeenCalledWith(
      activeNote(notes[0].id, { ...notes[0] })
    );
  });
});
