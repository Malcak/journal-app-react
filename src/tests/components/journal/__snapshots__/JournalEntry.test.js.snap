// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <JournalEntry /> component should display correctly 1`] = `
<div
  className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
  onClick={[Function]}
>
  <div
    className="journal__entry-picture"
    style={
      Object {
        "backgroundImage": "url(https://res.cloudinary.com/malcak-media-cloud/image/upload/v1626378813/htjjxjjndcweyzl1zesr.jpg)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      Hello World
    </p>
    <p
      className="journal__entry-content"
    >
      And welcome to a new video...
    </p>
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
      Thursday
    </span>
    <h4>
      15th
    </h4>
  </div>
</div>
`;
